### ISSUE TITLE
Allow audio input channel selection

### ISSUE THREAD

**Issue by canuckistani** [obsproject/obs-studio#2556]
## Expected Behavior

If I select an audio interface with multiple audio channels, I should be able to select which channels I want as the audio source.

## Current Behavior

Currently it seems like you can only get the first two input channels as a stereo pair. If one wants to route different channels into OBS you need to use additional software like loopback or Audio Hijack to route the audio.

## Additional information

The use case is to support more complex audio setups, for example my [DJ mixer](https://www.allen-heath.com/ahproducts/xone96/) outputs the master output to channels 11/12, and in order to stream a DJ set I need to use a combination of Audio Hijack and Blackhole to route the audio into OBS. It feels like the UI in the Audio Input Properties dialog would be a great place to allow the user to select which channel pair to get audio from.

**Comment by WizardCM** [comment]
The OBS Studio GitHub issue tracker is **only** to be used for reporting bugs that have replication steps. New features should be discussed on our Discord (https://obsproject.com/discord/) or Ideas page (https://ideas.obsproject.com/). For support issues, use the Discord or forums (https://obsproject.com/forum/)

Additionally, this is purely based on how your mixer works - for a proper multi-channel device (I use 5.1 speakers) they will get downmixed to stereo in OBS (if that's what OBS is set to). If you want to rearrange channels for inputs, try the [rematrix filter](https://obsproject.com/forum/resources/rematrix-filter-plugin.620/).

**Comment by canuckistani** [comment]
Thanks for getting back to me.

It's great to hear that a plugin could add in this functionality into OBS, however rematrix filter doesn't seem to support macOS or be actively maintained. The workaround works for me as-is, and I will post something to the ideas page.

**Comment by canuckistani** [comment]
@WizardCM I commented the ideas site on the existing idea that seemed most relevant:

https://ideas.obsproject.com/posts/199/multichannel-audio-interface-support-for-mac

I also looked at rematrix filter as suggested; someone recently contributed macOS builds but it doesn't seem to help in this specific use case needed by users of OBS on macOS who want to route audio from Allen & Heath mixers. The use case again is simply being able to route audio from input channels of a hardware USB audio interface that aren't inputs 1&2.

In the meantime, there have been a number of posts on social media with questions on how to do this. Windows users are happy with using the ASIO utility but Mac users end up wrestling with various workarounds that might affect the sound quality and stability of their live stream, ranging from expensive third party utilities to RCA cable loopbacks.
